simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp

```diff
@@ -367,8 +373,396 @@ TEST(TrafficLight, TrafficLight)
+    EXPECT_TRUE(bulb.is(Status::flashing));
+    EXPECT_TRUE(bulb.is(Shape::circle));
+    EXPECT_TRUE(bulb.is(Shape::Category::circle));
+  }
+
+  {
+    const auto bulb = Bulb("red flashing");
+
+    EXPECT_TRUE(bulb.is(Color::red));
+    EXPECT_TRUE(bulb.is(Status::flashing));
+    EXPECT_TRUE(bulb.is(Shape::circle));
+    EXPECT_TRUE(bulb.is(Shape::Category::circle));
+  }
+
+  {
+    const auto bulb = Bulb("red");
+
+    EXPECT_TRUE(bulb.is(Color::red));
+    EXPECT_TRUE(bulb.is(Status::solid_on));
+    EXPECT_TRUE(bulb.is(Shape::circle));
+    EXPECT_TRUE(bulb.is(Shape::Category::circle));
+  }
+
+  {
+    const auto bulb = Bulb("green solidOn right");
+
+    EXPECT_TRUE(bulb.is(Color::green));
+    EXPECT_TRUE(bulb.is(Status::solid_on));
+    EXPECT_TRUE(bulb.is(Shape::right));
+    EXPECT_TRUE(bulb.is(Shape::Category::arrow));
+  }
+
+  {
+    const auto bulb = Bulb("green right");
+
+    EXPECT_TRUE(bulb.is(Color::green));
+    EXPECT_TRUE(bulb.is(Status::solid_on));
+    EXPECT_TRUE(bulb.is(Shape::right));
+    EXPECT_TRUE(bulb.is(Shape::Category::arrow));
+  }
+}
+
+/**
+ * @note Test basic functionality. Test whether the TrafficLight message
+ * is constructed configured according to the Bulb object.
+ */
+TEST(TrafficLight, Bulb_trafficLightMessageConversion)
+{
+  using TrafficLight = traffic_simulator::TrafficLight;
+  using Color = TrafficLight::Color;
+  using Status = TrafficLight::Status;
+  using Shape = TrafficLight::Shape;
+  using Bulb = TrafficLight::Bulb;
+
+  {
+    constexpr auto bulb = Bulb(Color::red, Status::flashing, Shape::circle);
+    std::ostringstream oss;
+    oss << bulb;
+    EXPECT_TRUE(oss.str() == "red flashing circle");
+  }
+
+  {
+    constexpr auto bulb = Bulb(Color::green, Status::solid_on, Shape::right);
+    std::ostringstream oss;
+    oss << bulb;
+    EXPECT_TRUE(oss.str() == "green solidOn right");
+  }
+
+  {
+    const auto bulb = Bulb("red flashing circle");
+    std::ostringstream oss;
+    oss << bulb;
+    EXPECT_TRUE(oss.str() == "red flashing circle");
+  }
+
+  {
+    const auto bulb = Bulb("red flashing");
+    std::ostringstream oss;
+    oss << bulb;
+    EXPECT_TRUE(oss.str() == "red flashing circle");
+  }
+
+  {
+    const auto bulb = Bulb("red");
+    std::ostringstream oss;
+    oss << bulb;
+    EXPECT_TRUE(oss.str() == "red solidOn circle");
+  }
+
+  {
+    const auto bulb = Bulb("green solidOn right");
+    std::ostringstream oss;
+    oss << bulb;
+    EXPECT_TRUE(oss.str() == "green solidOn right");
+  }
+
+  {
+    const auto bulb = Bulb("green right");
+    std::ostringstream oss;
+    oss << bulb;
+    EXPECT_TRUE(oss.str() == "green solidOn right");
+  }
+}
+
+/**
+ * @note Test basic functionality. Test function behavior when called with invalid name.
+ */
+TEST(TrafficLight, Color_make_wrong)
+{
+  using Color = traffic_simulator::TrafficLight::Color;
+  Color color;
+  EXPECT_THROW(color = Color::make("wrong_color"), std::runtime_error);
```

Comment:
Assessment seems unnecessary. Isn't the exception called in the `make` function?